src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -142,30 +130,10 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
                     epoch_with_gap: EpochWithGap::new(self.read_epoch, MAX_SPILL_TIMES),
                 };
                 self.iterator.seek(full_key.to_ref()).await?;
-                // We check `is_valid` and `user_key_out_of_range` here to avoid `delete_range_iter.seek`
-                // as much as possible because it can be expensive. We can simplify this logic after delete
-                // range is deprecated and replaced with vnode watermark.
-                if !self.iterator.is_valid() {
```

Comment:
@hzxa21 Do we still need to keep this check?

Reply:
It will be handled in the later `try_advance_to_next_valid`. Previously we early check this to avoid unnecessary delete iter seek.